### PR TITLE
Make Cors filters run before any other authorization filters

### PR DESCRIPTION
**Pull request by kichalla**
@rynowak

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Cors/CorsAuthorizationFilter.cs [hunk 1]
```diff
@@ -41,7 +41,9 @@ public int Order
         {
             get
             {
-                return int.MaxValue - 100;
+                // Since clients' preflight requests would not have data to authenticate requests, this
+                // filter must run before any other authorization filters.
+                return int.MinValue + 100;
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Cors/CorsAuthorizationFilter.cs]: Isn't this going to overflow?

Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.Cors/CorsAuthorizationFilter.cs]: Its `min` (-2,147,483,648) and not `max`

**Comment by davidfowl** [comment]
There's no better way to implement this? What about a resource filter?

**Comment by kichalla** [comment]
@davidfowl Authorization filters run before the resource filters

**Comment by rynowak** [comment]
> There's no better way to implement this? 

This is what order is for bro. Both of these are policies that tell you if you're allowed to access the action.

**Comment by davidfowl** [comment]
@rynowak then what about using an order relative to the thing you're trying to go before?

File: test/WebSites/CorsWebSite/Startup.cs [hunk 2]
```diff
@@ -47,6 +47,26 @@ public void ConfigureServices(IServiceCollection services)
                                .WithMethods("PUT", "POST")
                                .WithExposedHeaders("exposed1", "exposed2");
                     });
+
+                options.AddPolicy(
+                    "AllowAll",
+                    builder =>
+                    {
+                        builder.AllowCredentials()
```
Comment (pranavkm) [inline test/WebSites/CorsWebSite/Startup.cs]: Use VS default formatting.

File: src/Microsoft.AspNet.Mvc.Cors/CorsAuthorizationFilter.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Cors/CorsAuthorizationFilter.cs]: :+1:

**Comment by kichalla** [comment]
ping @rynowak @pranavkm

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**